### PR TITLE
HARVESTER: Fix build harvester pkg

### LINKED ISSUE #7160: Allow use of aliases and scoped packages in imports within all the folders
# Description

Allow to use aliases and scoped packages in imports in e2e and other paths.

Occurrences:
- ~@rak-phillip `@rancher/components` in [E2E tests](https://github.com/rancher/dashboard/actions/runs/2922903499/jobs/4660940044)~ solved with rancher/dashboard#7425
- @n313893254 `@pkg` [when using unit tests](https://github.com/rancher/dashboard/actions/runs/3334501675/jobs/5517530061) 

## Context

Due to a proper configuration in any other folder out of `./shell`, it is not possible to use aliases or scoped packages. This forces the use of relative paths which is quite inconvenient.
E.g.:
- [In `pkg/harvester`](https://github.com/rancher/dashboard/blob/master/pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachineVolume/type/existing.vue#L9) `import { HCI } from '../../../../types';`

### PR DESCRIPTION
**Pull request by n313893254**
### Summary
Fixes Harvester `yarn build-pkg harvester` failed
Fixes #7160

### Occurred changes and/or fixed issues

- Fix can not find `@/pkg/harvester` alias, change to `@pkg/harvester`.
- Fix can not find `@shell/types` in Harvester pkg
- Fix `yarn run test:ci` error, update `jest.config.js`

### CODE REVIEW

**Comment by n313893254** [comment]
Hi, @mantis-toboggan-md, PR ready, PTAL, Thx.

File: shell/pkg/vue.config.js [hunk 1]
```diff
@@ -53,6 +53,7 @@ module.exports = function(dir) {
       config.resolve.alias['./node_modules'] = path.join(maindir, 'node_modules');
       config.resolve.alias['@pkg'] = dir;
       config.resolve.alias['~pkg'] = dir;
+      config.resolve.alias[`@/pkg/${ pkgName }`] = dir;
```
Comment (richard-cox) [inline shell/pkg/vue.config.js:56]: This feels like a workaround, it's best to convert references from `@/pkg` to `@pkg`. There's some tech debt here as well, we (UI team) need to remove the `~pkg` alias (we have a gate that will fail if it's used)

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: shell/pkg/vue.config.js (hunk 1 above)
Comment (n313893254) [reply shell/pkg/vue.config.js]: Thanks for your review. Should I use `@pkg/harvester/config/labels-annotations` or `@pkg/config/labels-annotations`?  If I use `@pkg/config`, the unit test will be failed and throw an error that can not locale module @pkg/config/labels-annotations

https://github.com/rancher/dashboard/blob/07683d34a651d52416ebdbd90ec97135f0975116/jest.config.js#L15

Comment (n313893254) [reply shell/pkg/vue.config.js]: Thank you @richard-cox! Considering rancher-component package name is `@rancher/components`, I use the directory name as `pkgName`. 

And `'@pkg/(.*)': '<rootDir>/pkg/$1'` only works for `@pkg/harvester/config` not `@pkg/config`, it requires the `pkgName` to locale the directory path.

**Pull request merged**